src/background/tab-store.js

```diff
@@ -40,16 +40,7 @@ export default function TabStore(storage) {
     try {
       local = {};
       const loaded = JSON.parse(storage.getItem(key));
-      Object.keys(loaded).forEach(function (key) {
-        // ignore tab state saved by earlier versions of
-        // the extension which saved the state as a {key: <state string>}
-        // dict rather than {key: <state object>}
-        if (typeof loaded[key] === 'string') {
-          local[key] = { state: loaded[key] };
-        } else {
-          local[key] = loaded[key];
-        }
-      });
+      Object.keys(loaded).forEach(key => (local[key] = loaded[key]));
```

Comment:
It would be good to add a test to check what happens if the parsed JSON data is not an object. It looks like such code would always result in the fallback `local = local || {}` below initializing it to an empty object.

Reply:
I added a similar test in this other [PR](https://github.com/hypothesis/browser-extension/blob/d9cdf54ae6d669b842e915232d6c2ed2db6e43b3/tests/background/tab-store-test.js#L39)

Is this what you suggested?